ocaml/xapi/fileserver.ml

```diff
@@ -90,7 +90,13 @@ let send_file (uri_base : string) (dir : string) (req : Request.t)
   let s = Buf_io.fd_of bio in
   Buf_io.assert_buffer_empty bio ;
   if access_forbidden req s then
-    Http_svr.response_forbidden ~req s
+    match req.Request.host with
+    | Some host ->
+        (* Redirect towards HTTPS *)
+        let dest = "https://" ^ host ^ uri_base ^ req.Request.uri in
```

Comment:
Is this the best way to construct a URL? At least I would use `String.concat`, which is more efficient. Is there no way to systematically create a URL from components?

Reply:
Uri can be used here:
```ocaml
Uri.(make ~scheme:"https" ~host ~path:??? |> to_string)
```
https://docs.mirage.io/uri/Uri/index.html#val-make